js/scripts/util.js

```diff
@@ -173,10 +188,58 @@ const MUSIC = {
     LAVA_TENSE: { path: "./music/lava_tense.mp3", volume: 0.1 }, // might be timed task OR boss music
 }
 
+/**
+ * Gets an array of nearby entities from the midground.
+ * The entities must have a bounding box.
+ * 
+ * NOTE: I tried to optimize this as much as possible, but it still looks expensive with all the distance calculations.
+ * 
+ * @param {Vector} centerPos the central position from which to check for entities
+ * @param {number} range the radius of the circle to check for entities
+ * 
+ * @returns An array of entities within the range of the provided position
+ */
+const getNearbyEntities = (centerPos, range) => {
```

Comment:
Looks similar to some code I saw in other classes. This may be exactly what I was asking for in a prior comment to outsource checking if an entity is in view. If so than ignore my previous comment.